### ISSUE TITLE
BeltScanEvent

### ISSUE THREAD

**Issue by Syrrianyl** [EDCD/EDDI#59]
I know that it is some useless, but i work on it.

**in Event**
Create new BeltScannedEvent.cs

```
using EddiDataDefinitions;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Utilities;

namespace EddiEvents
{
    public class BeltScannedEvent : Event
    {
        public const string NAME = "Belt scanned";
        public const string DESCRIPTION = "Triggered when you complete a scan of a belt";
        public const string SAMPLE = @"{ ""timestamp"":""2017-09-05T08:35:15Z"", ""event"":""Scan"", ""BodyName"":""Synuefe RE-G c27-9 ABC A Belt Cluster 4"", ""DistanceFromArrivalLS"":223.959274}";
        public static Dictionary<string, string> VARIABLES = new Dictionary<string, string>();

        static BeltScannedEvent()
        {
            VARIABLES.Add("name", "The name of the belt that has been scanned");
			VARIABLES.Add("distancefromarrival", "The distance in LS from the main star");
        }

        public string name { get; private set; }

        public decimal distancefromarrival { get; private set; }

        public BeltScannedEvent(DateTime timestamp, string name, decimal distancefromarrival) : base(timestamp, NAME)
        {
            this.name = name;
            this.distancefromarrival = distancefromarrival;
        }

    }
}
```
add a line in EddiEvents.csproj:
`
    <Compile Include="BeltScannedEvent.cs" />`

**In DataDefinitions**

Create a Belt.cs
```

using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EddiDataDefinitions
{
    /// <summary>
    /// A star or planet
    /// </summary>
    public class Belt
    {
        /// <summary>The type of the body (Star or Planet or Belt?)</summary>
        public string type { get; set; }

        /// <summary>The name of the body</summary>
        public string name { get; set; }

        /// <summary>The name of the system in which the body resides</summary>
        public string systemname { get; set; }

        /// <summary>The distance of the body from the arrival star, in light seconds </summary>
        public long? distance { get; set; }
    }
}
```

Add a line in EddiDataDefinitions.csproj:
`    <Compile Include="Belt.cs" />`

**and in JournalMonitor.cs, make this change:**

```
                        case "Scan":
                            {
                                object val;
                                // Common items
                                string name = getString(data, "BodyName");
                                decimal distancefromarrival = getDecimal(data, "DistanceFromArrivalLS");

                                if (name.Contains("Belt Cluster"))
                                {

                                    events.Add(new BeltScannedEvent(timestamp, name, distancefromarrival) { raw = line });
                                    handled = true;
                                    break;
                                }
                                decimal radius = getDecimal(data, "Radius");
                                decimal? orbitalperiod = getOptionalDecimal(data, "OrbitalPeriod");
                                decimal rotationperiod = getDecimal(data, "RotationPeriod");
                                decimal? semimajoraxis = getOptionalDecimal(data, "SemiMajorAxis");
                                decimal? eccentricity = getOptionalDecimal(data, "Eccentricity");
                                decimal? orbitalinclination = getOptionalDecimal(data, "OrbitalInclination");
                                decimal? periapsis = getOptionalDecimal(data, "Periapsis");

```

Finally, in EDDI.json, add a new script:
```

    "Belt scanned": {
      "enabled": true,
      "priority": 3,
      "responder": true,
      "script": "{_ Context }\r\n{SetState('eddi_context_last_subject', 'belt')}\r\n{SetState('eddi_context_last_action', 'scan')}\r\n{SetState('eddi_context_body_system', system.name)}\r\n{SetState('eddi_context_body_name', event.name)}\r\n{Pause(1000)}\r\n\r\nScan of {P(event.name)} validated.\r\nJust another mere cluster belt.",
      "default": true,
      "name": "Belt scanned",
      "description": "Triggered when you complete a scan of a belt"
    },

```

(hope i forget nothing)

**Comment by Tkael** [comment]
I suppose this should work. Why not just revise the cottle script to recognize belt clusters? e.g. 

{if find(reportbody.name, "Belt Cluster") > 0:
   Belt cluster scanned.
|else:
    ...
}

**Comment by Hoodathunk** [comment]
As it stands, the 'Body scanned' EDDI event in the Journal Monitor is no longer filtering out 'Belt Cluster's and passing them through to all Responders, so a BeltScanEvent would be redundant.

At present, the belt cluster filtering has been moved to the Speech Responder, mainly because of deference to jgm's 'minimalist' philosophy. For reasons only known to jgm, he considered belt cluster scans 'garbage data'. I'm open to discussing the removal of the filter and passing through belt clusters for the 'Body scanned' cottle script to filter at the user's discretion.

Only minor issue is setting some 'Body scanned' event parameters to 'optional' in the associated event code ('radius', for example) due to exceptions created because those parameters are not present for belt clusters.

**Comment by Syrrianyl** [comment]
At the start I make that because I think that, perhabs later, we will have more interresting data in belt cluster that we can just had when it come

**Comment by Hoodathunk** [comment]
OK, I'm going to humbly eat my words. The 'fix' for 'Body scanned' event generating exceptions was significantly more convoluted & involved than originally suspected. The most elegant approach was to implement Syrrianyl's 'Belt scanned' code. Inserted BeltScanEvent.cs into EddiEvents project and pasted code in JournalMonitor.cs... worked 'right out the box', kudos to Syrrianyl. The Belt.cs definition was not used because the Body definition (Body.cs) worked just fine. I also added code to Eddi.cs to handle updating 'CurrentSystem' vairables and to the EDDN Responder to pass on the event. THe Speech Responder was also updated to filter for 'BeltScanEvent' events. Calling a Speech Responder script just for the name and 'Distance From Arrival' doesn't feel that compelling. Tested in-game, scanning some belts... worked great, no exceptions.
